sky/cli.py

```diff
@@ -1706,13 +1722,55 @@ def status(all: bool, refresh: bool, show_spot_jobs: bool, clusters: List[str]):
                           show_all=False,
                           limit_num_jobs_to_show=not all,
                           is_called_by_user=False))
-        click.echo(f'{colorama.Fore.CYAN}{colorama.Style.BRIGHT}Clusters'
-                   f'{colorama.Style.RESET_ALL}')
+        if ip:
+            if len(clusters) != 1:
+                with ux_utils.print_exception_no_traceback():
+                    plural = 's' if len(clusters) > 1 else ''
+                    cluster_num = (str(len(clusters))
+                                   if len(clusters) > 0 else 'No')
+                    raise ValueError(
+                        _STATUS_IP_CLUSTER_NUM_ERROR_MESSAGE.format(
+                            cluster_num=cluster_num,
+                            plural=plural,
+                            verb='specified'))
```

Comment:
How about we just handle this in L1751? May not be necessary to separate the two cases.

Reply:
There are some special cases...
For example:
* Say I have 3 clusters `test`, `cluster-1`, `cluster-2` in local database
* If we only check the length of `cluster_records` (the return value of `core.status`):
  * `sky status --ip test abc` will output the ip of cluster `test`, since the `_get_glob_clusters` doesn't find any clusters with `abc`
* If we only check the length of `clusters`
  * `sky status --ip 'cluster-*'` needs to throw an error since it will match two clusters